### PR TITLE
Add  me.njean.shootingpactris

### PR DESCRIPTION
**Pull request by Jehan**
Note: I've been discussing with the developer of Shooting Pactris by email as I was sad that his game was not easy to "install|run". I am mostly helping him out to get the first step quickly on flathub but will ask for him to have maintainance over the flathub repository (still waiting to get his github nickname, not sure if he has an account though).

Edit: original developer is @njean42 and he is ok to co-maintain this repository.

-----

Some parts of the manifest inspired by org.godotengine.godot.BaseApp
manifest, except that the BaseApp is managing Godot games the wrong way.
I don't want to provide a PCK as a source since it is pre-packaged data,
hence is bad for build transparency.
Instead, this manifest builds the game from the actual game source,
making it hence a much more trustable build. Also the finale binary does
not need Godot anymore.

Also adding a basic desktop file in here, but it will likely end up
upstream eventually as I want to help the developer a bit to package his
application (right now it's quite bare).

This flatpak is working fine (tested only on x86_64 so far, I'll tweak
the manifest on flathub to see how it will go on ARM) as we can confirm
after spending a good evening playing this fun game! :-)

### CODE REVIEW

File: me.njean.shootinpactris.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    "sdk": "org.freedesktop.Sdk",
+    "command": "shpactris",
+    "finish-args": [
+                        "--socket=x11",
+                        "--share=network",
+                        "--device=all",
+                        "--socket=pulseaudio"
+                    ],
+    "modules": [
+        "shared-modules/glu/glu-9.json",
```
Comment (bilelmoussaoui) [inline me.njean.shootinpactris.json:15]: You forgot to add shared-modules a git submodule

**Comment by Jehan** [comment]
Some more info for quick review: https://shootingpactris.njean.me/

The source: https://shootingpactris.njean.me

There is no appdata yet, but I'll work on that once the repo is in flathub (I'll probably also contribute this upstream directly).

File: me.njean.shootinpactris.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            ],
+            "build-options": {
+                "env": {
+                    "XDG_DATA_HOME": "/app/share/"
+                }
+            },
+            "build-commands": [
+                "/app/bin/godot-server --export 'Linux/X11' shpactris || true",
+                "install -Dm755 shpactris /app/bin/shpactris",
+                "install -Dm644 icon.png /app/share/icons/hicolor/64x64/apps/me.njean.shootingpactris.png",
```
Comment (bilelmoussaoui) [inline me.njean.shootinpactris.json:84]: We require a 128px icon as well

File: me.njean.shootinpactris.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                "install -Dm644 me.njean.shootingpactris.desktop /app/share/applications/me.njean.shootingpactris.desktop"
```
Comment (bilelmoussaoui) [inline me.njean.shootinpactris.json:85]: I guess the app doesn't ship with a metainfo file yet?

**Comment by bilelmoussaoui** [comment]
bot, build  me.njean.shootingpactris

**Comment by bilelmoussaoui** [comment]
bot, build me.njean.shootingpactris

File: me.njean.shootinpactris.json (hunk 1 above)
Comment (Jehan) [reply me.njean.shootinpactris.json:15]: Oups. Added!

**Comment by bilelmoussaoui** [comment]
bot, build me.njean.shootingpactris

File: me.njean.shootinpactris.json (hunk 2 above)
Comment (Jehan) [reply me.njean.shootinpactris.json:84]: Can it go in after the repository is done?

File: me.njean.shootinpactris.json (hunk 3 above)
Comment (Jehan) [reply me.njean.shootinpactris.json:85]: Not yet, no. I'll help the developer to have one.

File: me.njean.shootinpactris.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
+{
```
Comment (bilelmoussaoui) [inline me.njean.shootinpactris.json:1]: The manifest file should be named `me.njean.shootingpactris.json`

File: me.njean.shootinpactris.json [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    "app-id": "me.njean.shootingpactris",
+    "branch": "stable",
```
Comment (bilelmoussaoui) [inline me.njean.shootinpactris.json:3]:
```suggestion
```
It's already the default branch on Flathub

File: me.njean.shootinpactris.json (hunk 4 above)
Comment (Jehan) [reply me.njean.shootinpactris.json]: Oups! Typo now fixed.

File: me.njean.shootinpactris.json (hunk 5 above)
Comment (Jehan) [reply me.njean.shootinpactris.json]: Removed! I was just copy-pasting from our GIMP manifest. I guess I can remove the line from there as well then. :-)

**Comment by Jehan** [comment]
bot, build me.njean.shootingpactris

File: me.njean.shootinpactris.json (hunk 3 above)
Comment (bilelmoussaoui) [reply me.njean.shootinpactris.json]: It's a requirement for Flathub builds

File: me.njean.shootinpactris.json (hunk 2 above)
Comment (bilelmoussaoui) [reply me.njean.shootinpactris.json]: what do you mean? the 128px is what we display on the frontend and what other software centers/stores uses as well. You can either generate one upstream or up scale the current one

File: me.njean.shootinpactris.json (hunk 3 above)
Comment (Jehan) [reply me.njean.shootinpactris.json]: Ok. Let me do a quick one then in the flathub repo. I'll contribute it upstream later too.

**Comment by Jehan** [comment]
bot, build me.njean.shootingpactris

File: me.njean.shootinpactris.json (hunk 3 above)
Comment (Jehan) [reply me.njean.shootinpactris.json]: Done.

File: me.njean.shootingpactris.json [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                    "tag": "v1.0-online-beta2",
+                    "commit": "84d5e4c3976fbdd0ebe4eb55f7ec7bff8aba4fa1"
+                },
+                {
+                    "type": "file",
+                    "path": "me.njean.shootingpactris.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "me.njean.shootingpactris.appdata.xml"
```
Comment (LeandroStanger) [inline me.njean.shootingpactris.json:76]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: me.njean.shootingpactris.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    <binary>shpactris</binary>
+  </provides>
+
+  <releases>
+    <release version="1.0-online-beta3" date="2020-05-21">
+      <url type="details">https://framagit.org/nicooo/shpactris/-/tags/v1.0-online-beta3</url>
+      <description>
+        <_p>
+        Players can now chose a nickname when playing online.
+        </_p>
```
Comment (bilelmoussaoui) [inline me.njean.shootingpactris.appdata.xml:60]:
```suggestion
        <p>
        Players can now chose a nickname when playing online.
        </p>
```
This file isn't translated by intltools ?

Comment (Jehan) [reply me.njean.shootingpactris.appdata.xml]: Oups sorry about this. Bad copy-paste from a translated appdata file. Obviously this basic first version is not translated.

File: me.njean.shootingpactris.json (hunk 6 above)
Comment (Jehan) [reply me.njean.shootingpactris.json]: Fixed.

**Comment by Jehan** [comment]
bot, build me.njean.shootingpactris

**Comment by bilelmoussaoui** [comment]
The build fails due to the missing 128px version of the icon

File: me.njean.shootinpactris.json (hunk 2 above)
Comment (Jehan) [reply me.njean.shootinpactris.json]: Ok. Added some scaled-up and sharpened icon. It's ugly (the original 64x64 icon was not extraordinary itself!). But for a first step, it's cool.

**Comment by Jehan** [comment]
bot, build me.njean.shootingpactris

**Comment by Jehan** [comment]
The software developer answered that his github nick is @njean42 and he is good to be co-maintainer on the flatpak. So if you could please add him when this flatpak is validated, would be awesome!

**Comment by Jehan** [comment]
bot, build me.njean.shootingpactris

File: me.njean.shootingpactris.json [hunk 8]
```diff
@@ -0,0 +1,98 @@
+{
+    "app-id": "me.njean.shootingpactris",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "19.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "shpactris",
+    "finish-args": [
```
Comment (bilelmoussaoui) [inline me.njean.shootingpactris.json:7]: Nitpick: the finish-args indentations feels weird

**Comment by bilelmoussaoui** [comment]
Looks good, it would be nice to contribute the metainfo, desktop file & the icon upstream.

File: me.njean.shootingpactris.json (hunk 8 above)
Comment (Jehan) [reply me.njean.shootingpactris.json:7]: Indeed. Fixed.

**Comment by Jehan** [comment]
> Looks good, it would be nice to contribute the metainfo, desktop file & the icon upstream.

Of course. That was mostly some initial files to get the flatpak started. I will contribute these upstream and maybe these can be a bit more "finalized" when we do.

**Comment by Jehan** [comment]
Anything missing for this initial flathub build? 🙂

**Pull request closed without merging**